python/ray/data/preprocessor.py

```diff
@@ -226,59 +254,83 @@ def _transform(self, dataset: Dataset) -> Dataset:
             return dataset.map_batches(self._transform_pandas, batch_format="pandas")
-        elif transform_type == "arrow":
-            return dataset.map_batches(self._transform_arrow, batch_format="pyarrow")
+        elif transform_type == "numpy":
+            return dataset.map_batches(self._transform_numpy, batch_format="numpy")
         else:
             raise ValueError(
                 "Invalid transform type returned from _determine_transform_to_use; "
-                f'"pandas" and "arrow" allowed, but got: {transform_type}'
+                f'"pandas" and "numpy" allowed, but got: {transform_type}'
             )
 
-    def _transform_batch(self, df: "DataBatchType") -> "DataBatchType":
+    def _transform_batch(self, data: "DataBatchType") -> "DataBatchType":
         import pandas as pd
 
         try:
             import pyarrow
         except ImportError:
             pyarrow = None
 
-        if isinstance(df, pd.DataFrame):
+        if isinstance(data, pd.DataFrame):
             data_format = "pandas"
-        elif pyarrow is not None and isinstance(df, pyarrow.Table):
+        elif pyarrow is not None and isinstance(data, pyarrow.Table):
             data_format = "arrow"
+        elif isinstance(data, (dict, np.ndarray)):
+            data_format = "numpy"
         else:
             raise NotImplementedError(
-                "`transform_batch` is currently only implemented for Pandas DataFrames "
-                f"and PyArrow Tables. Got {type(df)}."
+                "`transform_batch` is currently only implemented for Pandas "
+                "DataFrames, PyArrow Tables, Numpy ndarray and dictionary of "
+                f"ndarary. Got {type(data)}."
             )
 
         transform_type = self._determine_transform_to_use(data_format)
 
         if transform_type == "pandas":
             if data_format == "pandas":
-                return self._transform_pandas(df)
-            else:
-                return self._transform_pandas(df.to_pandas())
-        elif transform_type == "arrow":
-            if data_format == "arrow":
-                return self._transform_arrow(df)
+                return self._transform_pandas(data)
             else:
-                return self._transform_arrow(pyarrow.Table.from_pandas(df))
+                return self._transform_pandas(data.to_pandas())
+        elif transform_type == "numpy":
+            if data_format == "numpy":
+                return self._transform_numpy(data)
+            elif data_format == "arrow":
+                if len(data.column_names) == 1:
+                    # If just a single column, return as a single numpy array.
+                    return self._transform_numpy(data[0].to_numpy())
+                else:
+                    output_dict = {}
+                    for col_name in data.column_names:
+                        output_dict[col_name] = data[col_name].to_numpy()
+                return self._transform_numpy(output_dict)
```

Comment:
Indentation, since this return should be part of the else block.
```suggestion
                    return self._transform_numpy(output_dict)
```